src/background/background.js

```diff
@@ -66,11 +49,32 @@ async function load() {
 
     await addMenuItemProcessInbox();
     await addMenuItemProcessFolder();
+};
 
-    // Add a listener for the onNewMailReceived events.
-    // On each new message decide what to do
-    // Messages are already filtered by junk classification and message filters
-    await messenger.messages.onNewMailReceived.addListener(onNewMailReceived);
-}
+// listener creation has to be on top level to work with manifest v3 
+// https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers#register-listeners
```

Comment:
I am confused why this is a chrome page.  
Thunderbird is based on firefox.

EDIT: Just saw that WebExtension should be a unified API. Still there are differences between chrome and firefox.
https://blog.mozilla.org/addons/2022/05/18/manifest-v3-in-firefox-recap-next-steps/

Reply:
yes, the behavior of the service worker/background task is not very well described in the Firefox documentation, so I have linked to the Chrome documentation.